### PR TITLE
Added small performance improvements

### PR DESCRIPTION
**Pull request by TheRealHaui**
Added small performance improvements.

Signed-off-by: Michael Hausegger <hausegger.michael@googlemail.com>

### CODE REVIEW

File: jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java [hunk 1]
```diff
@@ -725,7 +725,7 @@ public HttpCookie getSessionCookie(HttpSession session, String contextPath, bool
         if (isUsingCookies())
         {
             String sessionPath = (_cookieConfig.getPath()==null) ? contextPath : _cookieConfig.getPath();
-            sessionPath = (sessionPath==null||sessionPath.length()==0) ? "/" : sessionPath;
+            sessionPath = (sessionPath==null|| sessionPath.isEmpty()) ? "/" : sessionPath;
```
Comment (olamy) [inline jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java]: might be a bit nitpicking but do not add space here

Comment (TheRealHaui) [reply jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java]: As long as this is the only problem ...
No problem!
;-)
Done.

**Comment by joakime** [comment]
> In fact, we could probably do with a StringUtil method `isNullOrEmpty` to clean up a but more.

`StringUtil.isBlank(String)` exists ...

https://github.com/eclipse/jetty.project/blob/3af45f4c89adf000db6396f7b253d9b528cac301/jetty-util/src/main/java/org/eclipse/jetty/util/StringUtil.java#L435-L472

**Comment by gregw** [comment]
@joakime `isBlank` is different to `isEmpty`, as white space is blank but not empty.

**Comment by TheRealHaui** [comment]
@grew,
well, of course you are right - we do not know what the JIT does or even might do in the future.
Even more, thinking of various different JREs.

However, looking at the implementations of String#isEmpty() and/or different List#isEmpty() implementations we see mostly "== 0" statements which from a top level perspective I generally consider to be more performant than smaller or greater comparisons. Furthermore, using isEmpty() keeps one on the safe side independent of the JIT and from my personal point of view additionally makes more clear what the purpose of the line of code is.
Finally, of course I could create microbenchmark tests however I do trust others when the statements are so clear: https://stackoverflow.com/questions/1508975/why-is-list-size0-slower-than-list-isempty-in-java

**Comment by gregw** [comment]
No need for a micro benchmark, I was more interested if you had read something somewhere, as often there are surprises (eg [this](https://medium.com/@hanru.yeh/collection-toarray-new-t-0-will-be-faster-than-collection-toarray-new-t-size-5b129fabfa88)).  Besides `isEmpty` reads better.       Do you want to create a `StringUtil.isEmpty(String)` to include the null check or shall I merge as is?

**Comment by TheRealHaui** [comment]
@gregw,
thanks for your reply!
Yes, completely agree with you regarding surprises.
That's why I tend regarding this topics to believe others.
At least when it is on Stackoverflow and without discussions and/or opossing opinions.

I am going to add the StringUtil.isEmpty(String) method if you want.
No problem.
However, just to clarify: Under isEmpty() == true for a String I understand: It is null or "".
Nothing else.

**Comment by TheRealHaui** [comment]
"However, just to clarify: Under isEmpty() == true for a String I understand: It is null or ""." which is logically wrong. My mistake. Should have thought more in detail about it before answering.
However, in addition there is already String#isEmpty() in the JDK.
Do you simply want to have an additional wrapper in the StringUtil class?

**Comment by olamy** [comment]
I did a bit of testing in the branch jetty-9.4.x_isempty_jmh
Locally using JDK 11.0.2, OpenJDK 64-Bit Server VM, 11.0.2+9

```
Benchmark                                    Mode  Cnt           Score           Error  Units
StringIsEmptyBenchmark.longIsEmpty          thrpt   45  4883444384.047 ± 340467687.126  ops/s
StringIsEmptyBenchmark.longIsEmptyLength    thrpt   45  5015885215.342 ± 370991750.834  ops/s
StringIsEmptyBenchmark.mediumIsEmpty        thrpt   45  5552816683.725 ± 180580885.127  ops/s
StringIsEmptyBenchmark.mediumIsEmptyLength  thrpt   45  5413860932.920 ± 249431576.460  ops/s
StringIsEmptyBenchmark.shortIsEmpty         thrpt   45  5376170110.478 ± 247502232.023  ops/s
StringIsEmptyBenchmark.shortIsEmptyLength   thrpt   45  4639732481.324 ± 218147071.673  ops/s
```
isEmpty use ` str == null || str.isEmpty() `
IsEmptyLength use ` str == null || str.length() == 0 `
strings length are short 12, medium 90, long 295 

Test run on a load instance:
https://jenkins.webtide.net/job/load_testing/job/jmh_runs_9.4.x/89/jmh-run-report/

I cannot see a very huge difference except for short size string.

**Comment by gregw** [comment]
isEmpty looks better and is a little faster! let's go with it.
thanks

**Pull request merged**